File: 3. Week 3 - Line Chart/exercise/r/visualization.R
# Visualize Trailing 365-day Distance using ggplot2

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
library(ggplot2)
library(readr)

# We need to make sure the path points to our data file
data_path <- "../../data/trailing_365_distance.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # Ensure Date parsing (readr usually handles YYYY-MM-DD well)
  # But good to be explicit if needed: df$Date <- as.Date(df$Date)
  
  # Plot
  # TODO: Create a line chart using ggplot2
  # Use geom_line()
  # Consider formatting the x axis for dates using scale_x_date
  
  # p <- ggplot(df, aes(x = Date, y = TotalDistance)) + ...
  
  
  # Uncomment to save
  # ggsave("trailing_distance_chart.png", plot = p, width = 12, height = 6)
  # print("Chart saved to trailing_distance_chart.png")

} else {
  print(paste("Error: File not found at", data_path))
}

File: 3. Week 3 - Line Chart/exercise/python/visualization.py
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib.dates as mdates
import os

def visualize():
    data_path = '../../data/trailing_365_distance.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    # TODO: Load data and convert Date column to datetime objects
    # df = ...
    # df['Date'] = ...
    
    # 1. Plot
    plt.figure(figsize=(12, 6))
    
    # TODO: Plot the line chart
    # plt.plot(...)
    
    # 2. Formatting
    # TODO: Add title and labels
    # plt.title(...)
    # plt.xlabel(...)
    # plt.ylabel(...)
    plt.grid(True, linestyle='--', alpha=0.5)
    
    # 3. Date Axis Formatting
    # TODO: Format the x-axis to show Year-Month and use YearLocator
    # plt.gca().xaxis.set_major_formatter(...)
    # plt.gca().xaxis.set_major_locator(...)
    plt.xticks(rotation=45)
    
    plt.legend()
    plt.tight_layout()
    
    # TODO: Save the chart
    # plt.savefig(...)
    
    print("Chart saved to trailing_distance_chart.png")
    plt.show()

if __name__ == "__main__":
    visualize()
